numba/core/untyped_passes.py

```diff
@@ -626,109 +627,123 @@ def run_pass(self, state):
             calls = [_ for _ in blk.find_exprs('call')]
             for call in calls:
-                glbl = guard(get_definition, func_ir, call.func)
-                if glbl and isinstance(glbl, (ir.Global, ir.FreeVar)):
-                    # find a literal_unroll
-                    if glbl.value is literal_unroll:
-                        if len(call.args) > 1:
-                            msg = "literal_unroll takes one argument, found %s"
-                            raise errors.UnsupportedError(msg % len(call.args),
-                                                          call.loc)
-                        # get the arg, make sure its a build_list
-                        unroll_var = call.args[0]
-                        to_unroll = guard(get_definition, func_ir, unroll_var)
-                        if (isinstance(to_unroll, ir.Expr) and
-                                to_unroll.op == "build_list"):
-                            # make sure they are all const items in the list
-                            for i, item in enumerate(to_unroll.items):
-                                val = guard(get_definition, func_ir, item)
-                                if not val:
-                                    msg = ("multiple definitions for variable "
-                                           "%s, cannot resolve constant")
-                                    raise errors.UnsupportedError(msg % item,
-                                                                  to_unroll.loc)
-                                if not isinstance(val, ir.Const):
-                                    msg = ("Found non-constant value at "
-                                           "position %s in a list argument to "
-                                           "literal_unroll" % i)
-                                    raise errors.UnsupportedError(msg,
-                                                                  to_unroll.loc)
-                            # The above appears ok, now swap the build_list for
-                            # a built tuple.
-
-                            # find the assignment for the unroll target
-                            to_unroll_lhs = guard(get_definition, func_ir,
-                                                  unroll_var, lhs_only=True)
-
-                            if to_unroll_lhs is None:
+                glbl = None
+                glbl_tmp = guard(get_definition, func_ir, call.func)
+                if isinstance(glbl_tmp, ir.Expr) and glbl_tmp.op == "getattr":
+                    if glbl_tmp.attr == "literal_unroll":
+                        # resolve the getattr, make sure it's getting
+                        # attr "literal_unroll" from the numba module.
+                        got_attr = guard(get_definition, func_ir,
+                                         glbl_tmp.value)
+                        if isinstance(got_attr, (ir.Global, ir.FreeVar)):
```

Comment:
should there be a `else` branch that raise unreachable error? i can't imagine `got_attr` being anything but `(ir.Global, ir.FreeVar)`?